client/rpc/block.go

```diff
@@ -2,57 +2,68 @@ package rpc
 
 import (
 	"context"
+	"encoding/hex"
 	"fmt"
-	"strconv"
-
-	"github.com/spf13/cobra"
+	"strings"
+	"time"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/cosmos/cosmos-sdk/codec/legacy"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tm "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
 )
 
-// BlockCommand returns the verified block data for a given heights
-func BlockCommand() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "block [height]",
-		Short: "Get verified data for the block at given height",
-		Args:  cobra.MaximumNArgs(1),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx, err := client.GetClientQueryContext(cmd)
-			if err != nil {
-				return err
-			}
-			var height *int64
-
-			// optional height
-			if len(args) > 0 {
-				h, err := strconv.Atoi(args[0])
-				if err != nil {
-					return err
-				}
-				if h > 0 {
-					tmp := int64(h)
-					height = &tmp
-				}
-			}
-
-			output, err := getBlock(clientCtx, height)
-			if err != nil {
-				return err
-			}
-
-			fmt.Println(string(output))
-			return nil
-		},
+// get the current blockchain height
+func GetChainHeight(clientCtx client.Context) (int64, error) {
+	node, err := clientCtx.GetNode()
+	if err != nil {
+		return -1, err
 	}
 
-	cmd.Flags().StringP(flags.FlagNode, "n", "tcp://localhost:26657", "Node to connect to")
+	status, err := node.Status(context.Background())
+	if err != nil {
+		return -1, err
+	}
 
-	return cmd
+	height := status.SyncInfo.LatestBlockHeight
+	return height, nil
 }
 
-func getBlock(clientCtx client.Context, height *int64) ([]byte, error) {
+// QueryBlocksByEvents performs a search for blocks for a given set of events
+// via the Tendermint RPC. An event takes the form of:
+// "{eventAttribute}.{attributeKey} = '{attributeValue}'". Each event is
+// concatenated with an 'AND' operand. It returns a slice of Info object
+// containing blocks and metadata. An error is returned if the query fails.
+// If an empty string is provided it will order blocks by asc
+func QueryBlocksByEvents(clientCtx client.Context, events []string, page, limit int, orderBy string) (*sdk.SearchBlocksResult, error) {
+	// XXX: implement ANY
+	query := strings.Join(events, " AND ")
```

Comment:
Recall, we're not supposed to parse any events. Just directly pass the query to Tendermint

Reply:
The input `events []string` needs to be translated to a string for the rpc call... It doesn't accept an array.  @alexanderbez 
I can do `query := strings.Join(events, "")` but not sure that's correct

Reply:
Yeah, it needs to be the way it is apparently. the `And` is required by the RPC - it would be nice to have the options of `AND` `OR` in the future.